### PR TITLE
Added annotation that InitialResources set request

### PR DESCRIPTION
**Pull request by piosz**
addresses #13589

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
This seems like a generally useful pattern that I may also use it for ```LimitRange```.

File: plugin/pkg/admission/initialresources/admission.go [hunk 1]
```diff
@@ -109,14 +110,23 @@ func (ir initialResources) estimateAndFillResourcesIfNotSet(pod *api.Pod) {
 			c.Resources.Requests = api.ResourceList{}
 			req = c.Resources.Requests
 		}
+		annotation := ""
 		if cpu != nil {
 			glog.Infof("CPU estimation for container %v in pod %v/%v is %v", c.Name, pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, cpu.String())
+			annotation = "cpu"
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/initialresources/admission.go]: any reason you don't just do ```annotation = string(api.ResourceCPU)```

File: plugin/pkg/admission/initialresources/admission.go [hunk 2]
```diff
@@ -109,14 +110,23 @@ func (ir initialResources) estimateAndFillResourcesIfNotSet(pod *api.Pod) {
...
 			req[api.ResourceCPU] = *cpu
 		}
 		if mem != nil {
 			glog.Infof("Memory estimation for container %v in pod  %v/%v is %v", c.Name, pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, mem.String())
+			if annotation != "" {
+				annotation += ", "
+			}
+			annotation += "memory"
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/initialresources/admission.go]: same here: ```annotation = string(api.ResourceMemory)``` - makes it easier to find who references resources by name.

File: plugin/pkg/admission/initialresources/admission.go [hunk 3]
```diff
@@ -109,14 +110,23 @@ func (ir initialResources) estimateAndFillResourcesIfNotSet(pod *api.Pod) {
 			c.Resources.Requests = api.ResourceList{}
 			req = c.Resources.Requests
 		}
+		annotation := ""
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/initialresources/admission.go]: actually, why not make this a slice ```annotations := []string{}``, append each resource to slice, and at end, you can just do ```strings.Join(annotations, ",")```... this makes it easier to manage going forward with more resources and less error prone to re-ordering if blocks.

File: plugin/pkg/admission/initialresources/admission.go [hunk 4]
```diff
@@ -109,14 +110,23 @@ func (ir initialResources) estimateAndFillResourcesIfNotSet(pod *api.Pod) {
...
 		if mem != nil {
 			glog.Infof("Memory estimation for container %v in pod  %v/%v is %v", c.Name, pod.ObjectMeta.Namespace, pod.ObjectMeta.Name, mem.String())
+			if annotation != "" {
+				annotation += ", "
+			}
+			annotation += "memory"
 			req[api.ResourceMemory] = *mem
 		}
+		if annotation != "" {
+			pod.ObjectMeta.Annotations[initialResourcesAnnotation] = annotation + " request set by InitialResources plugin"
```
Comment (derekwaynecarr) [inline plugin/pkg/admission/initialresources/admission.go]: If I have more than one container in a pod, and only some of those containers get their resource requirements set, should we state which container had its request set?

**Comment by derekwaynecarr** [comment]
fwiw, i made a similar change to LimitRanger here: https://github.com/kubernetes/kubernetes/pull/13998  would be good if we can settle on common pattern.  for example, if we should or should not label which container in a pod was set or not.

**Comment by piosz** [comment]
Thank you @derekwaynecarr for comments. I'll follow your pattern.

**Comment by piosz** [comment]
We definitely want to have information for different containers. How about:
*Initial Resources plugin set: cpu, memory request for container c1; cpu request for container c4; memory request for container c6.*

@derekwaynecarr WDYT?

**Comment by derekwaynecarr** [comment]
@piosz - I think your proposed text looks good.  I will update my PR to follow similar pattern.

File: plugin/pkg/admission/initialresources/admission.go (hunk 1 above)
Comment (piosz) [inline plugin/pkg/admission/initialresources/admission.go]: Done

File: plugin/pkg/admission/initialresources/admission.go (hunk 2 above)
Comment (piosz) [inline plugin/pkg/admission/initialresources/admission.go]: Done

File: plugin/pkg/admission/initialresources/admission.go (hunk 3 above)
Comment (piosz) [inline plugin/pkg/admission/initialresources/admission.go]: Done

File: plugin/pkg/admission/initialresources/admission.go (hunk 4 above)
Comment (piosz) [inline plugin/pkg/admission/initialresources/admission.go]: As proposed below

**Comment by piosz** [comment]
@derekwaynecarr, I changed the code to follow the proposed pattern. PTAL

**Comment by piosz** [comment]
@derekwaynecarr friendly ping

**Comment by derekwaynecarr** [comment]
Lgtm

**Comment by piosz** [comment]
Thanks for review!

**Comment by piosz** [comment]
@k8s-bot test this please

**Comment by piosz** [comment]
@k8s-bot test this please

**Pull request merged**